packages/@react-native-windows/telemetry/src/telemetry.ts

```diff
@@ -4,207 +4,411 @@
+   * @param _contextObjects An optional context object.
+   * @returns Whether to kee
+   */
+  private static basicTelemetryProcessor(
+    envelope: appInsights.Contracts.EnvelopeTelemetry,
+    _contextObjects?: {
+      [name: string]: any;
+    },
+  ): boolean {
+    delete envelope.tags['ai.cloud.roleInstance'];
+    return true;
+  }
+
+  /**
+   * Performs the processing necessary (mostly PII sanitization) for error events.
+   * @param envelope
+   * @param _contextObjects
+   * @returns
+   */
+  private static errorTelemetryProcessor(
+    envelope: appInsights.Contracts.EnvelopeTelemetry,
+    _contextObjects?: {
+      [name: string]: any;
+    },
+  ): boolean {
+    if (envelope.data.baseType === 'ExceptionData') {
+      const data = envelope.data.baseData;
+      if (data?.exceptions) {
+        for (const exception of data.exceptions) {
+          for (const frame of exception.parsedStack) {
+            errorUtils.sanitizeErrorStackFrame(frame);
+          }
+          // CodedError has non-PII information in its 'type' member, plus optionally some more info in its 'data'.
+          // The message may contain PII information. This can be sanitized, but for now delete it.
+          // Note that the type of data.exceptions[0] is always going to be ExceptionDetails. It is not the original thrown exception.
+          // https://github.com/microsoft/ApplicationInsights-node.js/issues/707
+          if (Telemetry.options.preserveErrorMessages) {
+            exception.message = errorUtils.sanitizeErrorMessage(
+              exception.message,
+            );
+          } else {
+            delete exception.message;
+          }
+        }
+      }
     }
-    Telemetry.preserveMessages = preserveMessages ? true : false;
+    return true;
+  }
 
-    if (process.env.RNW_CLI_TEST) {
-      Telemetry.client.commonProperties.isTest = process.env.RNW_CLI_TEST;
+  /** Tries to update the version of the named package/tool by calling getValue(). */
+  static async tryUpdateVersionsProp(
+    name: string,
+    getValue: () => Promise<string | null>,
+    forceRefresh?: boolean,
+  ): Promise<boolean> {
+    if (!Telemetry.client) {
+      return true;
     }
-    if (!Telemetry.client.commonProperties.sessionId) {
-      Telemetry.client.commonProperties.sessionId = randomBytes(16).toString(
-        'hex',
-      );
-      Telemetry.client.addTelemetryProcessor(sanitizeEnvelope);
+
+    if (forceRefresh === true || !Telemetry.versionsProp[name]) {
+      const value = await getValue();
+      if (value) {
+        Telemetry.versionsProp[name] = value;
+        return true;
+      }
     }
+    return false;
   }
 
-  static isCI(): boolean {
-    return (
-      process.env.AGENT_NAME !== undefined || // Azure DevOps
-      process.env.CIRCLECI === 'true' || // CircleCI
-      process.env.TRAVIS === 'true' || // Travis
-      process.env.CI === 'true' // other CIs
+  /** Populates the versions property of tools we care to track. */
```

Comment:
No, we don't break down how any of the time is spent running a command.

Reply:
It might be an interesting metric to start gathering times for phases of a command, but I don't think it needs to be in this initial check-in. https://github.com/microsoft/react-native-windows/issues/9109